server/protocols/protocols-pop3/src/main/java/org/apache/james/pop3server/mailbox/MailboxAdapter.java

```diff
@@ -1,174 +1,175 @@
-            if (results.hasNext()) {
-                return results.next().getHeaders().getInputStream();
-            } else {
-                return null;
-            }
-        } catch (MailboxException e) {
-            throw new IOException("Unable to retrieve message header for uid " + uid, e);
-        } finally {
-            mailboxManager.endProcessingRequest(session);
-        }
-    }
-
-    @Override
-    public InputStream getMessage(String uid) throws IOException {
-        try {
-            mailboxManager.startProcessingRequest(session);
-            Iterator<MessageResult> results = manager.getMessages(MessageUid.of(Long.parseLong(uid)).toRange(), FULL_GROUP, session);
-            if (results.hasNext()) {
-                return results.next().getFullContent().getInputStream();
-            } else {
-                return null;
-            }
-        } catch (MailboxException e) {
-            throw new IOException("Unable to retrieve message for uid " + uid, e);
-        } finally {
-            mailboxManager.endProcessingRequest(session);
-        }
-    }
-
-    @Override
-    public List<MessageMetaData> getMessages() throws IOException {
-        try {
-            mailboxManager.startProcessingRequest(session);
-            Iterator<MessageResult> results = manager.getMessages(MessageRange.all(), METADATA_GROUP, session);
-            List<MessageMetaData> mList = new ArrayList<>();
-            while (results.hasNext()) {
-                MessageResult result = results.next();
-                MessageMetaData metaData = new MessageMetaData(String.valueOf(result.getUid().asLong()), result.getSize());
-                mList.add(metaData);
-            }
-            return ImmutableList.copyOf(mList);
-        } catch (MailboxException e) {
-            throw new IOException("Unable to retrieve messages", e);
-        } finally {
-            mailboxManager.endProcessingRequest(session);
-        }
-    }
-
-    @Override
-    public void remove(String... uids) throws IOException {
-        List<MessageUid> uidList = Arrays.stream(uids)
-            .map(uid -> MessageUid.of(Long.parseLong(uid)))
-            .collect(Guavate.toImmutableList());
-
-        try {
-            mailboxManager.startProcessingRequest(session);
-            manager.delete(uidList, session);
-        } catch (MailboxException e) {
-            String serializedUids = uidList
-                .stream()
-                .map(uid -> uid.toString())
-                .collect(Collectors.joining(",", "[", "]"));
-
-            throw new IOException("Unable to remove messages: " + serializedUids, e);
-        } finally {
-            mailboxManager.endProcessingRequest(session);
-        }
-    }
-
-    @Override
-    public String getIdentifier() throws IOException {
-        try {
-            mailboxManager.startProcessingRequest(session);
-            long validity = manager.getMailboxEntity()
-                    .getUidValidity();

```

Comment:
AFAICT, `getIdentifier` method result is never used ...

Reply:
It is used by UidlCmdHandler